### PR TITLE
Feat/pill component

### PR DESCRIPTION
**Pull request by themonster2015**
## Description

Create the Pill and Snackbar components.
## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/design-system-core/src/components/Icon/index.tsx [hunk 1]
```diff
@@ -18,6 +18,7 @@ export interface IconProps {
   disabled?: boolean;
   testId?: string;
   styling?: string;
+  onClick?: () => void;
```
Comment (didd) [inline ui/design-system-core/src/components/Icon/index.tsx]:
```suggestion
  onClick?: (event: React.SyntheticEvent<Element, Event>) => void;
```

File: ui/design-system-core/src/components/Snackbar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+export type snackBarType = 'alert' | 'caution' | 'success' | 'info';
+
+export interface ISnackbar {
+  title?: string;
+  description?: string;
+  actionButtonLabel?: string;
+  type: snackBarType;
+  handleButtonClick?: () => void;
+  handleDismiss?: () => void;
```
Comment (didd) [inline ui/design-system-core/src/components/Snackbar/index.tsx]:
```suggestion
  handleDismiss?: (event: React.SyntheticEvent<Element, Event>) => void;
```

File: ui/design-system-core/src/components/Snackbar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import { getColorLight, getColorDark } from './getColor';
+
+export type snackBarType = 'alert' | 'caution' | 'success' | 'info';
+
+export interface ISnackbar {
+  title?: string;
+  description?: string;
+  actionButtonLabel?: string;
+  type: snackBarType;
+  handleButtonClick?: () => void;
```
Comment (didd) [inline ui/design-system-core/src/components/Snackbar/index.tsx]:
```suggestion
  handleButtonClick?: (event: React.SyntheticEvent<Element, Event>) => void;
```

Comment (themonster2015) [reply ui/design-system-core/src/components/Snackbar/index.tsx]: thanks Didd 👍

**Review by quininez — COMMENTED** [review]
you have conflicts with next that need to be fixed

File: ui/design-system-core/src/components/Icon/index.tsx [hunk 4]
```diff
@@ -18,6 +18,7 @@ export interface IconProps {
   disabled?: boolean;
   testId?: string;
   styling?: string;
+  onClick?: (event: React.SyntheticEvent<Element, Event>) => void;
```
Comment (didd) [inline ui/design-system-core/src/components/Icon/index.tsx]:
```suggestion
```

Let's keep the API of the icon component lean and clean. You can handle the `onClick` event as follows:

```
<button onClick={...}>
    <Icon .../>
</button>
```

Once I finish the work on enhancing the button component I will replace the above suggestion with something like :

```
<Button onClick={...}>
    <Icon .../>
</Button>
```

It's a better API design to wrap every design system component which needs the `onClick` handler with the button element or component. That way we maintain the semantic meaning of each component. Buttons handle click events and the other components address a specific concern. In the case of icons the concern is simply to render icons of different sizes and styles.

Comment (jenvonau) [reply ui/design-system-core/src/components/Icon/index.tsx]: Thanks, Didd. I have made the change.

Comment (themonster2015) [reply ui/design-system-core/src/components/Icon/index.tsx]: Thanks, Didd. I have made the change.

Comment (quininez) [reply ui/design-system-core/src/components/Icon/index.tsx]: But why is it a better design like this? Is there any specific advantage of keeping semantic purity that outweighs the hassle of having to wrap components like divs or icons  whenever you want a simple click handler? @didd

Comment (didd) [reply ui/design-system-core/src/components/Icon/index.tsx]: @quininez what makes it better API design is to gain the benefits of better SEO and accessibility as a result of using semantic HTML. As a result of using button we can gain a native support for keyboard navigation, better support for screen readers, we can get a native support for disabled state of an element, better unit testing experience and the like. 

And in this PR's case we have an icon button which we can use but the button component will need to be enhanced first then we can have a choice of using the onClick handler as 
`<Button icon={...} onClick={...} ...` or 
`<Button onClick={...}> <Icon .../> </Button> `

File: ui/design-system-core/src/components/Snackbar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      </span>
+      <div className={tw('w-11/12')}>
+        <Text variant="button-md" color={tw(textcolor)}>
+          {title}
+        </Text>
+        <Text variant="body2" color={tw(textcolor)}>
+          {description}
+        </Text>
+        {actionButtonLabel && (
+          <span onClick={handleButtonClick} className={tw(`cursor-pointer`)}>
```
Comment (didd) [inline ui/design-system-core/src/components/Snackbar/index.tsx]:
```suggestion
          <button onClick={handleButtonClick}>
```

Latter on I will change it to the button component `Button`. You can check [this](https://www.builder.io/blog/buttons) and [this](https://javascript.plainenglish.io/stop-using-divs-for-buttons-87a0b3d7945e) links to get a better insight on why we prefer button and anchor tag over other elements for handling click events.

File: ui/design-system-core/src/components/Icon/index.tsx (hunk 4 above)
Comment (themonster2015) [reply ui/design-system-core/src/components/Icon/index.tsx]: That's a nice explanation Didd 👍

**Review by quininez — DISMISSED** [review]

**Review by quininez — APPROVED** [review]

**Pull request merged**